migrations/V14__create_foreign_key_constraint_on_billing_status_table.sql

```diff
@@ -0,0 +1,5 @@
+ALTER TABLE billing.billing_status
+  ADD CONSTRAINT event_id_fkey FOREIGN KEY (event_id) REFERENCES billing.billing_events(event_id)
+   ON DELETE RESTRICT
```

Comment:
I always get this confused. I know this foreign key constraint on the `billing_status` table _references_ the `billing_events` table. Which I _think_ translates as [per the docs](https://www.postgresql.org/docs/9.5/ddl-constraints.html) into `billing_status` containing "referencing" rows, and `billing_events` containing "referenced" rows, right? But I have difficulty figuring out how that maps onto what the constraint is actually doing... Does it mean:

- if a row in `billing_events` would be deleted, the deletion is prevented if there is a matching `event_id` in `billing_status`
- if a row in `billing_status` would be deleted, the deletion is prevented if there is a matching `event_id` in `billing_events`
- both
- neither / something else?

Reply:
The constraint is basically saying you can only create a `billing_status` if the `event_id` exists in `billing_events`.

The `ON DELETE RESTRICT` modifier will prevent deletion of the record from `billing_events` if corresponding `billing_status` record(s) exists.

The `ON UPDATE CASCADE` modifier states that if the `event_id` is modified in `billing_events` then the corresponding rows `billing_status` record will also be updated.

Both modifiers are probably unnecessary in our case as I can't envisage scenarios where we would be modifying `event_id` or deleting `billing_events` records, but I always like to be explicit as to the expected behaviour as the default behaviour would be to throw an error if an `UPDATE` or `DELETE` would result in the data integrity be compromised.

Reply:
Great. Thank you for explaining. 👍